dipy/align/vector_fields.pyx

```diff
@@ -3099,71 +3158,87 @@ def _gradient_3d(floating[:, :, :] img, double[:, :] img_world2grid,
     inside : array, shape (S', R', C')
         the buffer in which to store the flags indicating whether the sample
         point lies inside (=1) or outside (=0) the image grid
+    num_threads : int
+        Number of threads. If None (default) then all available threads
+        will be used.
     """
     cdef:
-        int nslices = out.shape[0]
-        int nrows = out.shape[1]
-        int ncols = out.shape[2]
-        int i, j, k, in_flag
-        double tmp
-        double[:] x = np.empty(shape=(3,), dtype=np.float64)
-        double[:] dx = np.empty(shape=(3,), dtype=np.float64)
-        double[:] h = np.empty(shape=(3,), dtype=np.float64)
-        double[:] q = np.empty(shape=(3,), dtype=np.float64)
+        int nslices = out.shape[0], k
+        int all_cores = openmp.omp_get_num_procs()
+        int threads_to_use = -1
+
+    if num_threads is not None:
```

Comment:
`threads_to_use = num_threads if num_threads is not None else all_cores`
but `sklearn` rather set `-1` as definition of all cores and replace `None` by `1`